mesosclient/testing/fake_mesos.go

```diff
@@ -105,16 +108,26 @@ func (fm *FakeMesos) getTasks() *master.Response_GetTasks {
 	for _, task := range fm.tasks {
 		switch stateLifecycle[*task.State] {
 		case stateActive:
-			appendTask(&getTasks.Tasks, task)
+			appendTask(&getTasks.Tasks, copyTask(task))
 		case stateTerminated:
-			appendTask(&getTasks.CompletedTasks, task)
+			appendTask(&getTasks.CompletedTasks, copyTask(task))
 		case stateUnreachable:
-			appendTask(&getTasks.UnreachableTasks, task)
+			appendTask(&getTasks.UnreachableTasks, copyTask(task))
 		}
 	}
 	return &getTasks
 }
 
+// copyTask serialises and deserialises a task in order to deep-copy it.
```

Comment:
🤔 Sneaky..

Reply:
This is my standard deep-copy mechanism for anything that supports it. It's probably more expensive than some others, but if the serialisation is self-contained (and protobuf's is) it's guaranteed to not accidentally leave mutable references to other things or whatever. It may strip out any fields that don't get serialised and/or change some internal representations (which would result in a non-identical copy), but in this case that doesn't happen.